### PR TITLE
Add missing beacon API eventstream-types

### LINKED ISSUE #5152: Add missing beacon API eventstream types
### Describe the feature

Add missing rpc-types for beacon event types

API definitions:

https://github.com/ethereum/beacon-APIs/blob/master/apis/eventstream/index.yaml

helpful tool: https://transform.tools/json-to-rust-serde

## TODO
* add type definitions
* add serde tests using sample data: `curl -X 'GET' 'http://localhost:5052/eth/v1/events?topics=payload_attributes' -H 'accept: text/event-stream'`, lighthouse docs: https://lighthouse-book.sigmaprime.io/api-bn.html#beacon-node-api
* add enum type `BeaconApiEvent{...}` and `BeaconApiEventTopic{...}`

### Additional context

payload attributes beacon api event type added here https://github.com/paradigmxyz/reth/pull/5130

### PR DESCRIPTION
**Pull request by i-m-aditya**
Fixes #5152

### CODE REVIEW

**Comment by i-m-aditya** [comment]
@mattsse Is the approach correct? I am receiving JSON string and then deserializing it to events struct.

File: examples/beacon-api-sse/src/main.rs
Comment (mattsse) [inline examples/beacon-api-sse/src/main.rs:1]: we want to limit this example to payload attributes only, so please revert changes here

File: examples/beacon-api-sse/src/main.rs [hunk 1]
```diff
@@ -52,6 +52,25 @@ struct BeaconEventsConfig {
...
+
+    #[arg(long = "topic", value_enum)]
+    pub event_topic: BeaconNodeEventTopic,
+}
+
+#[derive(Debug, Clone, clap::ValueEnum)]
+pub enum BeaconNodeEventTopic {
+    PayloadAtrributes,
+    Head,
+}
```
Comment (mattsse) [inline examples/beacon-api-sse/src/main.rs:64]: this can be moved to rpc-types crate

File: examples/beacon-api-sse/src/main.rs [hunk 2]
```diff
@@ -52,6 +52,25 @@ struct BeaconEventsConfig {
...
+
+impl BeaconNodeEventTopic {
+    fn to_string_representation(&self) -> String {
```
Comment (mattsse) [inline examples/beacon-api-sse/src/main.rs:67]:
```suggestion
    fn query_value(&self) -> &'static str {
```

**Review by mattsse — CHANGES_REQUESTED** [review]
nice start, this approach is correct, but we want to keep the example limited to payload attributes

we also want some tests

here's a sample for head

```json
{
    "slot": "202656",
    "block": "0x023b140bea768d741c42214a742bf6d50279828fd78774ebb077e8a7bd9cd7b9",
    "state": "0xa4ab3b5f3c00dff2842e545a683ecfe0190daea795fad62372f5f2c44fef40dd",
    "current_duty_dependent_root": "0x4b65252090bb84a831703f641f9abd36b27c5dc7a9b6d716d58122edbf6fc3de",
    "previous_duty_dependent_root": "0xa90c02c51761c1d20a7fff8f9ba403117ebd038be017304dbc2ab5d96b144064",
    "epoch_transition": true,
    "execution_optimistic": true
}
```

finalized checkpoint:

```json
{
    "block": "0xeac147a2c136eb4969f3c2c3b3447d438478befcb182082606af7162b504dd3f",
    "state": "0xbeeda7a8e3b3c9f4b6ba63cc6dad9f95024412f78fe33e323c019b14b9cdb819",
    "epoch": "6343",
    "execution_optimistic": true
}
```

**Review by mattsse — CHANGES_REQUESTED** [review]
this is great ty,

please remove all the `rename_all camelCase` attributes which then makes all the other renames redundant

**Comment by i-m-aditya** [comment]
> this is great ty,
> 
> please remove all the `rename_all camelCase` attributes which then makes all the other renames redundant

I was going to do it in draft PR, but missed it :/

File: crates/rpc/rpc-types/src/eth/engine/beacon_api/attestation.rs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Source {
+    pub epoch: String,
+    pub root: String,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/engine/beacon_api/attestation.rs:21]: serde attributes not needed

File: crates/rpc/rpc-types/src/eth/engine/beacon_api/attestation.rs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+pub struct Source {
+    pub epoch: String,
+    pub root: String,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/engine/beacon_api/attestation.rs:20]: add `Eq` derive

File: crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs [hunk 5]
```diff
@@ -7,6 +7,27 @@ use alloy_primitives::B256;
 use serde::{Deserialize, Serialize};
 use serde_with::{serde_as, DisplayFromStr};
 
+use super::{
+    attestation::AttestationData, light_client_finality::LightClientFinalityData,
+    light_client_optimistic::LightClientOptimisticData,
+};
+
+#[derive(Debug, Clone)]
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs:15]: Copy, PartialEq,Eq

File: crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs [hunk 6]
```diff
@@ -7,6 +7,27 @@ use alloy_primitives::B256;
...
+pub enum BeaconNodeEventTopic {
+    PayloadAtrributes,
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs:17]:
```suggestion
    PayloadAttributes,
```

File: crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs [hunk 7]
```diff
@@ -7,6 +7,27 @@ use alloy_primitives::B256;
 use serde::{Deserialize, Serialize};
 use serde_with::{serde_as, DisplayFromStr};
 
+use super::{
+    attestation::AttestationData, light_client_finality::LightClientFinalityData,
+    light_client_optimistic::LightClientOptimisticData,
+};
+
+#[derive(Debug, Clone)]
+pub enum BeaconNodeEventTopic {
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs:16]: we need a function that returns the topic name for the variant

**Review by mattsse — CHANGES_REQUESTED** [review]
this is great,

some requests, also we need to change all the string types.

epoch+slot as u64 and use serde_as DisplayFromStr
https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs#L207-L207
hash types need to be `B256`, like root,hash,commitment etc...

File: crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs [hunk 8]
```diff
@@ -28,6 +28,26 @@ pub enum BeaconNodeEventTopic {
     BlobSidecar,
 }
 
+impl BeaconNodeEventTopic {
+    pub fn query_value(&self) -> String {
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/engine/beacon_api/events.rs:32]:
```suggestion
    pub fn query_value(&self) -> &'static str {
```

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, last tiny nit

**Review by mattsse — APPROVED** [review]

**Pull request merged**